k8s/charts/seaweedfs/templates/filer-statefulset.yaml

```diff
@@ -209,7 +209,7 @@ spec:
               -s3.cert.file=/usr/local/share/ca-certificates/client/tls.crt \
               -s3.key.file=/usr/local/share/ca-certificates/client/tls.key \
               {{- end }}
-              {{- if .Values.filer.s3.allowEmptyFolder }}
+              {{- if eq (typeOf .Values.filer.s3.allowEmptyFolder) "bool" }}
```

Comment:
this does not seem a good fix. Better just define `.Values.filer.s3.allowEmptyFolder`.

Reply:
allowEmptyFolder is a boolean type, and it serves not only as a condition for evaluation but also as the value of a flag. However, when its value is false, Helm treats it as an empty value, and using an if condition will prevent it from being rendered.

The logic needed here should be that if allowEmptyFolder is either true or false, it should be rendered, while other empty values should not be.

the flag delault value is true
it can not be like this
```
if .Values.filer.s3.allowEmptyFolder
s3.allowEmptyFolder=true
else
s3.allowEmptyFolder=false
```
this is conflicted with default
maybe change the default value

Reply:
`allowEmptyFolder` default is false. The logic is correct?

Reply:
the default i mean is the flag's default in here,not chart default value
<img width="1034" alt="image" src="https://github.com/user-attachments/assets/173630e7-5ffe-4b0b-a71c-19be49f64772">
the flag default value is true,but chart default is false ,and now helm false no render,mean use flag default true,that's incorrect
if flag default is false ,chart value  false no render ,it's still false,the logic is correct